commons/Common.ml

```diff
@@ -747,7 +747,15 @@ let rec filter_some = function
   | None :: l -> filter_some l
   | Some e :: l -> e :: filter_some l
 
-let map_filter f xs = xs |> List.map f |> filter_some
+let map_filter f xs =
```

Comment:
maybe add a comment that says this version as opposed to old code: is tail recursive.